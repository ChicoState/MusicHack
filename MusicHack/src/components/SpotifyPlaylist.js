import React, {Component} from 'react';
import {StyleSheet, View, Text, Image} from 'react-native';

export default class SpotifyPlaylist extends Component {
  render() {
    return (
      <View style={styles.mainView}>
        <Text style={styles.textStyle}>
          {this.props.name}
        </Text>
        <Text style={styles.textStyle}>
          Total tracks: {this.props.tracks}
        </Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  mainView: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    borderWidth: 1,
    backgroundColor: '#aaa',
  },
  textStyle: {
    fontSize: 20,
    color: 'white',
    textShadowColor: '#44c',
    textShadowRadius: 5,
    marginBottom: 20,
  },
  imageView: {
    width: 300,
    height: 300,
  },
});
