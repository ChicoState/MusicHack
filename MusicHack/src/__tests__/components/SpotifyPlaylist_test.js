import React from 'react';
import SpotifyPlaylist from '../../components/SpotifyPlaylist';

import renderer from 'react-test-renderer';

beforeAll(() => {
  SpotifyPlaylist.prototype.componentDidMount = () => {};
});

test('renders correctly', () => {
  const tree = renderer.create(<SpotifyPlaylist />).toJSON();
  expect(tree).toMatchSnapshot();
});
