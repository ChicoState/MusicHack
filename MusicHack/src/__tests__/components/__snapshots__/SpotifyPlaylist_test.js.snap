// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#aaa",
      "borderWidth": 1,
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 20,
        "marginBottom": 20,
        "textShadowColor": "#44c",
        "textShadowRadius": 5,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 20,
        "marginBottom": 20,
        "textShadowColor": "#44c",
        "textShadowRadius": 5,
      }
    }
  >
    Total tracks: 
  </Text>
</View>
`;
